src/coreclr/vm/jithelpers.cpp

```diff
@@ -587,11 +587,43 @@ HCIMPL1_V(UINT32, JIT_Dbl2UIntOvf, double val)
 }
 HCIMPLEND
 
+HCIMPL1_V(UINT32, JIT_Dbl2UInt, double val)
+{
+    FCALL_CONTRACT;
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
```

Comment:
Same here. I'm not sure why the ifdef needs to exist.

We want the same behavior on all platforms. If the underlying platform behavior is already correct, then we won't be using this helper.